newlib/libc/misc/lock.c

```diff
@@ -81,7 +81,9 @@ subroutines are required for linking multi-threaded applications.
 
 /* dummy lock routines and static locks for single-threaded apps */
 
-#ifndef __SINGLE_THREAD__
+#include <newlib.h>
+
+#if defined(__SINGLE_THREAD__)
```

Comment:
This is actually a bug.
These are the dummy implementation if user selected `_RETARGETABLE_LOCKING` but it is still a `SINGLE_THREAD` configuration.